src/gui/MainWindow.cpp

```diff
@@ -1284,16 +1284,40 @@ void MainWindow::processTrayIconTrigger()
     }
 }
 
+#ifndef _WIN32
+void MainWindow::show()
+{
+    m_lastShowTime = Clock::currentMiliSecondsSinceEpoch();
+    QMainWindow::show();
+}
+
+bool MainWindow::shouldHide()
+{
+    qint64 current_time = Clock::currentMiliSecondsSinceEpoch();
+
+    if (current_time - m_lastShowTime < 50)
+        return false;
+
+    return true;
+}
+
+void MainWindow::hide()
+{
+    if (shouldHide())
+        QMainWindow::hide();
```

Comment:
braces.

Reply:
like braces around every if, even oneline?

Reply:
yes that is the style for our code base